public/pages/Rules/components/RuleEditor/components/YamlRuleEditorComponent/YamlRuleEditorComponent.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState } from 'react';
+import { load } from 'js-yaml';
+import { EuiFormRow, EuiCodeEditor, EuiLink, EuiSpacer, EuiText, EuiCallOut } from '@elastic/eui';
+import FormFieldHeader from '../../../../../../components/FormFieldHeader';
+import { Rule } from '../../../../../../../models/interfaces';
+import {
+  mapRuleToYamlObject,
+  mapYamlObjectToYamlString,
+  mapYamlObjectToRule,
+} from '../../../../utils/mappers';
+
+export interface YamlRuleEditorComponentProps {
+  rule: Rule;
+  change: React.Dispatch<Rule>;
+  isInvalid: boolean;
+  errors?: string[];
+}
+
+export interface YamlEditorState {
+  errors: string[] | null;
+  value?: string;
+}
+
+export const YamlRuleEditorComponent: React.FC<YamlRuleEditorComponentProps> = ({
+  rule,
+  change,
+  isInvalid,
+  errors,
+}) => {
+  const yamlObject = mapRuleToYamlObject(rule);
+
+  const [state, setState] = useState<YamlEditorState>({
+    errors: null,
+    value: mapYamlObjectToYamlString(yamlObject),
+  });
+
+  const onChange = (value: string) => {
+    setState((prevState) => ({ ...prevState, value }));
+  };
+
+  const onBlur = () => {
+    if (!state.value) {
+      setState((prevState) => ({ ...prevState, errors: ['Rule cannot be empty'] }));
+      return;
+    }
+    try {
+      const yamlObject = load(state.value);
+
+      const rule = mapYamlObjectToRule(yamlObject);
+
+      change(rule);
+
+      setState((prevState) => ({ ...prevState, errors: null }));
+    } catch (error) {
+      setState((prevState) => ({ ...prevState, errors: ['Invalid YAML'] }));
+
+      console.warn('Security Analytics - Rule Eritor - Yaml load', error);
+    }
+  };
+
+  const renderErrors = () => {
+    if (state.errors && state.errors.length > 0) {
```

Comment:
rewrite as `state.errors?.length > 0`

Reply:
not possible like this way if I've understood right?
<img width="486" alt="image" src="https://user-images.githubusercontent.com/1564068/210049672-abde0102-94c4-422a-826f-b8d7e95e3bde.png">

Reply:
aah, leave it as is then